addons/binding/org.openhab.binding.network/src/main/java/org/openhab/binding/network/service/NetworkService.java

```diff
@@ -29,159 +31,276 @@
-                }
-            } catch (SocketTimeoutException se) {
-                logger.debug("timed out while connecting to host '{}' port '{}' timeout '{}'",
-                        new Object[] { hostname, port, timeout });
-            } catch (IOException ioe) {
-                logger.debug("couldn't establish network connection [host '{}' port '{}' timeout '{}']",
-                        new Object[] { hostname, port, timeout });
-            } catch (InterruptedException e) {
-                logger.debug("ping program was interrupted");
+    public void startAutomaticRefresh(ScheduledExecutorService scheduledExecutorService) {
+        cancelRefreshJob();
+        refreshJob = scheduledExecutorService.scheduleWithFixedDelay(new Runnable() {
+            @Override
+            public void run() {
+                performPresenceDiscovery();
             }
+        }, 0, refreshIntervalInMS, TimeUnit.MILLISECONDS);
+
+        enableDHCPListen(useDHCPsniffing);
+    }
 
-        } while (currentTry++ < this.retry);
+    public void stopAutomaticRefresh() {
+        cancelRefreshJob();
+        enableDHCPListen(false);
+    }
 
-        return -1;
+    /**
+     * Enables/Disables listing for dhcp packets to figure out if devices have entered the network. This does not work
+     * for iOS devices. The hostname of this network service object will be registered to the dhcp request packet
+     * listener if enabled and unregistered otherwise.
+     *
+     * @param enabled Enable/Disable the dhcp listen service for this hostname.
+     */
+    private void enableDHCPListen(boolean enabled) {
+        try {
+            if (enabled) {
+                try {
+                    if (ReceiveDHCPRequestPackets
+                            .register(InetAddress.getByName(hostname).getHostAddress(), updateListener)
+                            .isUseUnprevilegedPort()) {
+                        dhcpState = "No access right for port 67. Bound to port 6767 instead. Port forwarding necessary!";
+                    }
+                } catch (SocketException e) {
+                    logger.warn("Cannot use DHCP sniffing.", e);
+                    useDHCPsniffing = false;
+                    dhcpState = "Cannot use DHCP sniffing: " + e.getLocalizedMessage();
+                }
+            } else {
+                ReceiveDHCPRequestPackets.unregister(InetAddress.getByName(hostname).getHostAddress());
+                dhcpState = "off";
+            }
+        } catch (UnknownHostException e) {
+            logger.warn("Hostname invalid: {}", hostname, e);
+        }
     }
 
-    @Override
-    public String toString() {
-        return this.hostname + ";" + this.port + ";" + this.retry + ";" + this.refreshInterval + ";" + this.timeout
-                + ";" + this.useSystemPing;
+    /**
+     * Return true if this device is recognised as IOS device
+     */
+    public boolean isIOSdevice() {
+        return true;
```

Comment:
I have no solution for this yet, maybe this should a configuration option for the binding or the thing. I'm pretty sure iOS devices can be recognized somehow, I just don't have one.